### PR TITLE
Feature/add service worker

### PR DESCRIPTION
**Pull request by OmgImAlexis**
This will only work on `http(s)://localhost` and all other HTTPS sites.
Self signed certs don't seem to work at all.

### CODE REVIEW

**Comment by p0psicles** [comment]
Sorry. What can we do with this?

**Comment by OmgImAlexis** [comment]
I'm using it to add proper caching to all assets. By using the service worker to look up the file first the browser can skip going to the server.

File: themes-default/slim/static/js/lib/globby-to-regexp.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
+const globbyToRegExp = function (glob, opts) {
```
Comment (sharkykh) [inline themes-default/slim/static/js/lib/globby-to-regexp.js]: Why not a minified file?

**Comment by sharkykh** [comment]
I'm pretty bummed that this doesn't work on self-signed certificates.
I keep reading that localhost should always be accepted as secure, but it never worked for me.

Seems like we currently have to choose between having push notifications (HTTPS on) or having the ServiceWorker (HTTPS off). :disappointed:

**Comment by labrys** [comment]
Disabling security should never be considered as an option.

**Comment by OmgImAlexis** [comment]
@sharkykh https://www.chromium.org/blink/serviceworker/service-worker-faq

>  I get an error message about "Only secure origins are allowed". Why?

> A: Service workers are only available to "secure origins" (HTTPS sites, basically) in line with a policy to prefer secure origins for powerful new features. However http://localhost is also considered a secure origin, so if you can, developing on localhost is an easy way to avoid this error.

**Comment by OmgImAlexis** [comment]
Ah... I'm testing on `http://localhost` and it works.

**Pull request closed without merging**